manifests/params.pp

```diff
@@ -351,7 +351,14 @@
 
   case $::operatingsystem {
     'Ubuntu': {
-      $server_service_provider = upstart
+      case $::lsbdistcodename {
+        'vivid': {
+          $server_service_provider = systemd
+        }
+        default: {
+          $server_service_provider = upstart
+        }
+      }
```

Comment:
and this whole thing can be replaced with a selector:
```puppet
$server_service_provider = $::lsbdistcodename {
     'vivid' => 'systemd',
     default: 'upstart'
}
```
and now the problem becomes extending it for future versions…

i'm really starting to feel the burden of the systemd breakage in debuntu… :(